packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -186,6 +186,24 @@ void exp(MachineState& m) {
     ++m.pc;
 }
 
+void signExtend(MachineState& m) {
+    m.stack.prepForMod(2);
+    auto& aNum = assumeInt(m.stack[0]);
+    auto& bNum = assumeInt(m.stack[1]);
+
+    if (aNum >= 32) {
```

Comment:
FYI, this could actually be >= 31, because the `aNum == 31` case is a no-op--you're extending beyond the top byte.  But the code as-is will give correct results.

Reply:
Ah actually there's a bug here from the flipping of the params

Reply:
The signextend impl is now cleaned up